### PR TITLE
Adding challenges

### PR DESCRIPTION
**Pull request by Vladmidir**
# Summary 
The admin can now add, delete and hide challenges

# Note
Must reset the db

# Screenshots
image
image
image

### CODE REVIEW

File: client/src/components/ChallengeCard.tsx [hunk 1]
```diff
@@ -2,15 +2,20 @@ import Card from "react-bootstrap/Card";
...
+import { ButtonToolbar } from "react-bootstrap";
 
 function ChallengeCard({
   title,
   generalDescription,
   id,
   difficulty,
-  completed
+  completed,
+  admin,
```
Comment (arthurhohinng) [inline client/src/components/ChallengeCard.tsx:14]: is `admin` a boolean? I think `isAdmin` would be a better variable name if it is

File: client/src/types/ChallengeDetailsShort.ts [hunk 2]
```diff
@@ -6,4 +6,6 @@ export type ChallengeDetailsShort = {
     generalDescription: string;
     id: number;
     completed: boolean;
+    admin: boolean; // Used for displaying the delete & hide buttons
```
Comment (arthurhohinng) [inline client/src/types/ChallengeDetailsShort.ts:9]: how about `isAdmin`?

File: client/src/pages/Challenges.tsx [hunk 3]
```diff
@@ -23,10 +25,34 @@ function Challenges() {
...
+  }, []);
+
+  // Fetch challenges from the server
   useEffect(() => {
-    //fetch data about the challenge with the provided "id"
     setIsLoading(true);
-    fetch("/api/challenges")
+
+    let url = "/api/challenges";
+    if (query.get("hidden") === "true") url = "/api/challenges/hidden";
```
Comment (arthurhohinng) [inline client/src/pages/Challenges.tsx:53]: so the challenges page url has a search parameter of `hidden`?

Comment (arthurhohinng) [reply client/src/pages/Challenges.tsx:53]: nvm i see it in the screenshot

**Review by arthurhohinng — APPROVED** [review]

**Comment by arthurhohinng** [comment]
Some minor comments but an additional thing i want to add is if we are reusing the challenges page for hidden challenges why not just have a button that allows you to toggle to view hidden challenges instead of adding an additional link in the navbar

However if you decide to keep it as a seperate page on the navbar you may want to make sure the link in the navbar is highlighted as it isn't highlighted in the screenshot

**Comment by Vladmidir** [comment]
> Some minor comments but an additional thing i want to add is if we are reusing the challenges page for hidden challenges why not just have a button that allows you to toggle to view hidden challenges instead of adding an additional link in the navbar
> 
> You could toggle and keep the current logic by having the button set search params of hidden to true
> 
> However if you decide to keep it as a seperate page on the navbar you may want to make sure the link in the navbar is highlighted as it isn't highlighted in the screenshot

I'd rather keep it as separate pages, because of how I implemented the backend. I will fix the highlighting, thank you!

**Pull request merged**